examples/LocalizationExample.cpp

```diff
@@ -84,15 +84,17 @@ class UnaryFactor: public NoiseModelFactor1<Pose2> {
   // The first is the 'evaluateError' function. This function implements the desired measurement
   // function, returning a vector of errors when evaluated at the provided variable value. It
   // must also calculate the Jacobians for this measurement function, if requested.
-  Vector evaluateError(const Pose2& q,
-                       boost::optional<Matrix&> H = boost::none) const override {
-    // The measurement function for a GPS-like measurement is simple:
-    // error_x = pose.x - measurement.x
-    // error_y = pose.y - measurement.y
-    // Consequently, the Jacobians are:
-    // [ derror_x/dx  derror_x/dy  derror_x/dtheta ] = [1 0 0]
-    // [ derror_y/dx  derror_y/dy  derror_y/dtheta ] = [0 1 0]
-    if (H) (*H) = (Matrix(2, 3) << 1.0, 0.0, 0.0, 0.0, 1.0, 0.0).finished();
+  Vector evaluateError(const Pose2& q, boost::optional<Matrix&> H = boost::none) const override {
+    // The measurement function for a GPS-like measurement h(q) which predicts the measurement (m) is h(q) = q, q = [qx qy qtheta]
+    // The error is then simply calculated as E(q) = h(q) - m:
+    // error_x = q.x - mx
+    // error_y = q.y - my
+    // Node's orientation reflects in the Jacobian, in tangent space this is equal to the right-hand rule rotation matrix
+    // H =  [ cos(q.theta)  -sin(q.theta) 0 ]
+    //      [ sin(q.theta)   cos(q.theta) 0 ]
+    const double cosTheta = cos(q.theta());
```

Comment:
same comments as above of course :-)

Reply:
Right! it is smart to use the available feature, I will update the PR soon.